File: src/main/java/com/naminhyeok/fantazzk/room/PlaceBidRequest.java
package com.naminhyeok.fantazzk.room;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

@Schema(description = "경매 입찰 요청")
record PlaceBidRequest(
    @Schema(description = "입찰 금액. 현재 최고가보다 높고 최소 증가폭을 만족해야 합니다.", example = "150")
    @NotNull(message = "입찰 금액은 필수입니다")
    @Positive(message = "입찰 금액은 1 이상이어야 합니다") Integer amount
) {
}

File: room/web/src/main/kotlin/com/naminhyeok/fantazzk/room/web/dto/PlaceBidRequest.kt
package com.naminhyeok.fantazzk.room.web.dto

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "경매 입찰 요청입니다.")
data class PlaceBidRequest(
    @field:Schema(description = "입찰을 수행할 팀장 ID 입니다. RoomResponse.teamLeaders[].id 값을 사용합니다.", example = "leader-02")
    val teamLeaderId: String,
    @field:Schema(description = "현재 최고가보다 큰 입찰 금액입니다.", example = "120")
    val amount: Int,
)
